### ISSUE TITLE
Allow duplicating instanced scene's children

### ISSUE THREAD

**Issue by snoopdouglas** [godotengine/godot-proposals#2203]
### Describe the project you are working on
Platformer (n/a)

### Describe the problem or limitation you are having in your project
I've instanced a scene, and have enabled 'Editable Children' on it.

I can now arbitrarily add children to the instanced scene, but am not allowed to duplicate one of its existing children:

![Screenshot from 2021-01-30 14-03-00](https://user-images.githubusercontent.com/4892574/106358434-0d5ab700-6304-11eb-8d1f-6f4dac9795cf.png)

Given that this produces a copy - and as such, doesn't interfere with the instanced scene's existing children (even if it is a shallow copy) - I'm wondering whether this limitation could be removed.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation
Go ahead and duplicate the child rather than showing an error.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
(not sure I can add anything here)

### If this enhancement will not be used often, can it be worked around with a few lines of script?
Possibly with a tool, though I'm not exactly sure how.

### Is there a reason why this should be core and not an add-on in the asset library?
In my opinion, it's useful enough to be in core.

**Comment by Xrayez** [comment]
The rationale may be that duplicating a child node from foreign scene may lead to having resources which are going to be shared between the new node and editable node (like if you duplicate collision shape nodes and the respective `Shape` resources will be used by all instances by default instead of being unique per instance), which in turn may break the idea behind having editable children which are more or less internal in nature. But most likely that the code simply forbids any kind of potentially mutable operations currently without discriminating other use cases.

**Comment by snoopdouglas** [comment]
Yeah, I agree that was probably the rationale. Duplication always produces a shallow copy though (ie. nodes' sub-resources become shared), so I don't think it makes a huge amount of difference if it happens to be a child of an instanced scene.

**Comment by KoBeWi** [comment]
Seems to have been implemented at some point (tested in 4.5).
